workshops/micro_bit_flappy_bird/README.md

```diff
@@ -0,0 +1,363 @@
+
+Don't mind about the previously existing code. 
+
+Create a new LED sprite pointing to the right. A sprite is like a little LED creature you can tell what to do.
+
+```js
+let bird: game.LedSprite = null
+bird = game.createSprite(0, 2)
+bird.set(LedSpriteProperty.Blink, 300)
+```
+
+A variable `bird` is declared in TypeScript. The syntax of declaring a varible in TypeScript is: `let identifier:data-type = value;`
+
+The `game.createSprite(x,y)` creates a sprite(our bird) in position (x, y) on the 5x5 grid of LEDs. Then we set our bird to blink every 300 milliseconds.
+
+You can see our bird flying like this:point_down:
+
+![blinking LED](https://cloud-70aevdanf.vercel.app/0bird.gif)
+
+## Step-2
+
+Now let's create some controls.  We’ll control the bird by pressing the A button to go up or the B button to go down.
+
+Micro:bit has built-in [event handlers](https://makecode.microbit.org/reference/event-handler) that are associated with a particular event, such as "button A pressed".  `.onButtonPressed` handler works when button A or B is pressed, or A and B together. They are just like event listeners in JavaScript but with custom events.
+
+```js
+input.onButtonPressed(Button.A, function () {
+    bird.change(LedSpriteProperty.Y, -1)
+})
+input.onButtonPressed(Button.B, function () {
+    bird.change(LedSpriteProperty.Y, 1)
+})
+```
+
+Now when button any is pressed, the position of the bird should change by 1 unit(up or down).
+
+![controls](https://cloud-70aevdanf.vercel.app/1controls.gif)
+
+## Step-3
+
+Let's create some randomly generated obstacles for the bird to crash💥
+
+We’ll keep all obstacles inside an array. All obstacles will have a single hole for the bird to fly through.
+
+First, create an array of `obstacles` that will hold all of the obstacle sprites.
+
+```js
+let obstacles: game.LedSprite[] = []
+```
+
+Now let's generate vertical obstacles consisting of 4 sprites and 1 random hole. Create a variable `emptyObstacleY` and store a random number between 0 and 4 in it. The function `randint(a,b)` creates a random number between `a` and `b`.
+
+```js
+let emptyObstacleY = 0
+let obstacles: game.LedSprite[] = []
+
+emptyObstacleY = randint(0, 4)
+```
+
+Using the `for` loop, iterate from 0 to 4. For every coordinate not equal to `emptyObstacleY` create and add obstacle sprites to the end of the obstacles array using the `push()` method.
+
+```js
+let emptyObstacleY = 0
+let obstacles: game.LedSprite[] = []
+
+emptyObstacleY = randint(0, 4)
+for (let index = 0; index <= 4; index++) {
+    if (index != emptyObstacleY) {
+        obstacles.push(game.createSprite(4, index))
+    }
+}
+```
+
+Now with every micro:bit restart you should see different autogenerated vertical obstacles.
+
+![random obstacles](https://cloud-70aevdanf.vercel.app/2random_obstacles.gif)
+
+Before continuing, make sure that obstacles are generated randomly and that the bird is moving up and down.
```

Comment:
also add a statement similar to "if it doesn't work as expected, try to cross check your code"